src/cpp/desktop/DesktopWebProfile.cpp

```diff
@@ -66,9 +66,23 @@ class Interceptor : public QWebEngineUrlRequestInterceptor
 
 } // end anonymous namespace
 
-WebProfile::WebProfile(const QUrl& baseUrl, QObject* parent)
-   : QWebEngineProfile(QString::fromUtf8("rstudio-desktop"), parent)
+WebProfile* WebProfile::getInstance(const QUrl &baseUrl)
 {
+   static WebProfile* instance = new WebProfile(baseUrl);
+   return instance;
+}
+
+WebProfile::WebProfile(const QUrl& baseUrl)
+   : QWebEngineProfile(QStringLiteral("rstudio"))
+{
+   // force an in-memory cache of web resources
+   // otherwise, multiple windows / pages attempting to access
+   // or mutate the disk cache at the same time can cause
+   // data corruption to occur
+   //
+   // https://github.com/rstudio/rstudio/issues/6201
+   setHttpCacheType(QWebEngineProfile::MemoryHttpCache);
```

Comment:
Given this code here, can we live without the static one-per-process WebProfile?

Reply:
Looks like you're correct. I've updated the PR (so now this is the only new code)